pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -159,6 +159,23 @@ func getPVSpecFromCache(name string, pvcReadOnly bool, expectedClaimUID types.UI
 	return volume.NewSpecFromPersistentVolume(clonedPV, pvcReadOnly), nil
 }
 
+// DetermineVolumeAction returns true if volume and pod needs to be added to dswp
+// and it returns false if volume and pod needs to be removed from dswp
+func DetermineVolumeAction(pod *v1.Pod, desiredStateOfWorld cache.DesiredStateOfWorld, defaultValue bool) bool {
+	if pod == nil || len(pod.Spec.Volumes) <= 0 {
+		return defaultValue
```

Comment:
In all cases if pod or spec has no volumes - the underlying operation is going to be a no-op. So in reality we don't really care what value we return from here.

But rather than hardcoding `true` or `false` from such branches - we let the caller or the function decide what he expects. For example - the default action to take in case of `podAdd` event is to attach the volume. So we let caller of `DetermineVolumeAction` supply a default action.  I implemented it this way to make sure - we are backward compatible as much as possible - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/attachdetach/attach_detach_controller.go#L388

As you can see from above line - the default behaviour on `podAdd` is `true`.